### PR TITLE
Refactor: Replace hard coded data

### PR DESCRIPTION
**Pull request by pauloiankoski**
Resolves [GIVE-300]
Depends on https://github.com/impress-org/givewp/pull/7233 then on https://github.com/impress-org/givewp/pull/7235

## Description

Since most of the front-end work has been completed during the discovery cycle, this pull request mainly involves replacing hardcoded data in our block and field templates with actual data loaded from Models.

PS: This work will need to be modified slightly once the changes from https://github.com/impress-org/givewp/pull/7227 are merged into the core. A new pull request will be created to include the updates.

## Testing Instructions
1. Add the new Event Tickets block to a page. Without any events into the database, a message including a link to add a new event must be displayed.
2.  Run the snippet bellow to add a sample Event.

```php
use Give\EventTickets\Models\Event;
use Give\EventTickets\Models\EventTicketType;
use Give\Framework\Support\ValueObjects\Money;
use Give\Framework\Support\Facades\DateTime\Temporal;

$event = Event::create([
    "title" => "My Event",
    "description" => "Description goes here",
    "startDateTime" => Temporal::toDateTime("2024-03-01 10:00:00"),
    "endDateTime" => Temporal::toDateTime("2024-03-01 16:00:00")
]);

EventTicketType::create([
    "eventId" => $event->id,
    "label" => "VIP",
    "description" => "Description goes here",
    "price" => Money::fromDecimal(100, give_get_currency()),
    "totalTickets" => 10
]);
```

3. Now that you have an event into the database, add an Event Tickets block to the page and now you may be able to select an event. Once selected, the block must reflect all data added into that sample event.
4. Publish the form and you must see the same when previewing the donation form.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

[GIVE-300]: https://stellarwp.atlassian.net/browse/GIVE-300?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/EventTickets/Models/EventTicketType.php [hunk 1]
```diff
@@ -147,6 +144,9 @@ public static function fromQueryBuilderObject($object): EventTicketType
         return new EventTicketType([
             'id' => (int)$object->id,
             'event_id' => (int)$object->event_id,
```
Comment (jonwaldstein) [inline src/EventTickets/Models/EventTicketType.php:146]: @pauloiankoski just a heads up, all our models use a camelCase convention for their property names so we'll want to follow that syntax for events 😄

Comment (pauloiankoski) [reply src/EventTickets/Models/EventTicketType.php:146]: @jonwaldstein I'll create another PR fixing that then I'll update this one.

Comment (pauloiankoski) [reply src/EventTickets/Models/EventTicketType.php:146]: https://github.com/impress-org/givewp/pull/7233

File: src/EventTickets/Repositories/EventTicketTypeRepository.php [hunk 2]
```diff
@@ -26,8 +24,6 @@ class EventTicketTypeRepository
      */
     private $requiredProperties = [
         'event_id',
-        'created_at',
-        'updated_at',
```
Comment (kjohnson) [inline src/EventTickets/Repositories/EventTicketTypeRepository.php:30]: Why did you remove `created_at` and `updated_at` as required fields? In the database migration for the Event Ticket Type table these columns are defined as `NOT NULL`.

File: src/EventTickets/Repositories/EventTicketTypeRepository.php [hunk 3]
```diff
@@ -64,15 +60,16 @@ public function insert(EventTicketType $eventTicketType)
         try {
             DB::table('give_event_ticket_types')
                 ->insert([
-                    'id' => $eventTicketType->id,
                     'event_id' => $eventTicketType->event_id,
                     'label' => $eventTicketType->label,
                     'description' => $eventTicketType->description,
-                    'price' => $eventTicketType->price->formatToMinorAmount(),
+                    'price' => $eventTicketType->price->formatToDecimal(),
```
Comment (kjohnson) [inline src/EventTickets/Repositories/EventTicketTypeRepository.php:66]: Currency values should be stored as the minor amount of the currency.

File: src/EventTickets/resources/components/EventTicketsList.tsx [hunk 4]
```diff
@@ -2,23 +2,28 @@ import EventTicketsListItem from './EventTicketsListItem';
...
                 return (
                     <EventTicketsListItem
-                        ticket={ticket}
-                        selectedTickets={selectedTickets[ticket.id]?.quantity ?? 0}
+                        ticketType={ticketType}
+                        selectedTickets={selectedTickets[ticketType.id]?.quantity ?? 0}
                         handleSelect={
-                            handleSelect ? handleSelect(ticket.id, ticket.quantity, ticket.price) : () => null
+                            handleSelect
+                                ? handleSelect(ticketType.id, ticketType.totalTickets, ticketType.price)
```
Comment (kjohnson) [inline src/EventTickets/resources/components/EventTicketsList.tsx:25]: Does `totalTickets` mean the total tickets allowed or the total tickets sold or the total tickets remaining?

File: src/EventTickets/resources/components/EventTicketsListItem.tsx [hunk 5]
```diff
@@ -30,7 +30,7 @@ export default function EventTicketsListItem({ticket, currency, selectedTickets,
                     </button>
                 </div>
                 <p className={'givewp-event-tickets__tickets__ticket__quantity__availability'}>
-                    {ticket.quantity - selectedTickets} {__('remaining', 'give')}
+                    {ticketType.totalTickets - selectedTickets} {__('remaining', 'give')}
```
Comment (kjohnson) [inline src/EventTickets/resources/components/EventTicketsListItem.tsx:33]: Does `totalTickets` mean the total tickets allowed or the total tickets available?

File: src/EventTickets/resources/templates/EventTickets/EventTicketsListHOC.tsx [hunk 6]
```diff
@@ -36,15 +36,15 @@ export default function EventTicketsListHOC({name, tickets, ticketsLabel, soldOu
...
 
-    const onSelectTicket: OnSelectTicketProps = (ticketId, ticketQuantity, ticketPrice) => (selectedQuantity) => {
+    const onSelectTicket: OnSelectTicketProps = (ticketId, totalTickets, ticketPrice) => (selectedQuantity) => {
         if (selectedQuantity < 0) {
             selectedQuantity = 0;
         }
 
-        if (selectedQuantity > ticketQuantity) {
-            selectedQuantity = ticketQuantity;
+        if (selectedQuantity > totalTickets) {
```
Comment (kjohnson) [inline src/EventTickets/resources/templates/EventTickets/EventTicketsListHOC.tsx:46]: Does `totalTickets` mean the total tickets allowed or the total tickets remaining?

File: src/EventTickets/Repositories/EventTicketTypeRepository.php (hunk 2 above)
Comment (pauloiankoski) [reply src/EventTickets/Repositories/EventTicketTypeRepository.php:30]: The validation method that uses this array is called before the `createdAt` and `updatedAt` values are even generated. If we don't pass those values when creating a new `EventTicketType`, it will trigger errors because the required properties haven't been provided. However, the expected behavior is for the system to automatically generate those values for us.

Comment (kjohnson) [reply src/EventTickets/Repositories/EventTicketTypeRepository.php:30]: Hm. Good point. If the validation is running before it is generated that makes sense.

File: src/EventTickets/Repositories/EventTicketTypeRepository.php (hunk 3 above)
Comment (pauloiankoski) [reply src/EventTickets/Repositories/EventTicketTypeRepository.php]: Reverted the change for now.

File: src/EventTickets/resources/components/EventTicketsList.tsx (hunk 4 above)
Comment (pauloiankoski) [reply src/EventTickets/resources/components/EventTicketsList.tsx:25]: All these totalTickets questions can only be answered once we define the `maxAvailable` question, including the new prop for sold tickets or number of available tickets.

File: src/EventTickets/Models/EventTicketType.php (hunk 1 above)
Comment (kjohnson) [reply src/EventTickets/Models/EventTicketType.php:146]: #7233 has been merged.

**Review by kjohnson — APPROVED** [review]
